src/azul/queues.py

```diff
@@ -79,34 +79,41 @@ def dump_all(self):
 
     def _dump(self, queue, path):
         logger.info('Writing messages from queue "%s" to file "%s"', queue.url, path)
+        messages = self.read_messages(queue)
+        self._dump_messages(messages, queue.url, path)
+        logger.info(f'Finished writing {path !r}')
+
+    def _dump_messages(self, messages, queue_url, path):
+        messages = [self._condense(message) for message in messages]
+        with write_file_atomically(path) as file:
+            content = {
+                'queue': queue_url,
+                'messages': messages
+            }
+            json.dump(content, file, indent=4)
+        logger.info('Wrote %i messages', len(messages))
+
+    def _get_messages(self, queue):
         messages = []
         while True:
             message_batch = queue.receive_messages(AttributeNames=['All'],
                                                    MaxNumberOfMessages=10,
                                                    VisibilityTimeout=300)
             if not message_batch:  # Nothing left in queue
-                break
+                return messages
             else:
                 messages.extend(message_batch)
-        self._dump_messages(messages, queue.url, path)
+
+    def read_messages(self, queue):
+        messages = self._get_messages(queue)
         message_batches = list(more_itertools.chunked(messages, 10))
         if self._delete:
             logger.info('Removing messages from queue "%s"', queue.url)
             self._delete_messages(message_batches, queue)
         else:
             logger.info('Returning messages to queue "%s"', queue.url)
             self._return_messages(message_batches, queue)
-        logger.info(f'Finished writing {path !r}')
-
-    def _dump_messages(self, messages, queue_url, path):
```

Comment:
Try to minimize the diff. The diff  looks like the _dump_messages method was moved unnecessarily.